modules/generator/registry/native_histogram.go

```diff
@@ -176,18 +174,8 @@ func (h *nativeHistogram) collectMetrics(appender storage.Appender, timeMs int64
 		// the same, and so Reset() call below can be used to clear the exemplars.
 		s.histogram = encodedMetric.GetHistogram()
 
-		// NOTE: A subtle point here is that the new histogram implementation is
-		// chosen in the registry.  This means that if the user overrides change
-		// from "both" to "classic", we will still be using the new histograms
-		// implementation. This is because the registry is not recreated when the
-		// overrides change. This is a tradeoff to avoid recreating the registry
-		// for every change in the overrides, but it does mean that even if user
-		// changes to "classic" histograms, we will still be using the new
-		// histograms implementation and want to avoid generating native
-		// histograms.
-
 		// If we are in "both" or "classic" mode, also emit classic histograms.
-		if overrides.HasClassicHistograms(h.modeFunc()) {
+		if overrides.HasClassicHistograms(h.histogramOverride) {
```

Comment:
Interesting side-effect from making the override a configuration setting. Just replacing the histogram all together instead of dynamically switching is a bit more overhead but I think it will be more reliable/sturdy.

Reply:
Right, I noticed this also.  Sturdy sounds right to me.